cli/cfncluster/cfncluster.py

```diff
@@ -125,58 +125,62 @@ def update(args):
-        resources = cfnconn.describe_stack_resources(stack_name)
-        while True:
-            temp_resources.extend(resources)
-            if not resources.next_token:
-                break
-            resources = cfnconn.describe_stack_resources(stack, next_token=resources.next_token)
-        resources = temp_resources
-        asg = [r for r in resources if r.logical_resource_id == 'ComputeFleet'][0].physical_resource_id
-
-        desired_capacity = asgconn.get_all_groups(names=[asg])[0].desired_capacity
+        resources = cfn.describe_stack_resources(StackName=stack_name).get('StackResources')
+        asg_name = [r for r in resources if r.get('LogicalResourceId') == 'ComputeFleet'][0].get('PhysicalResourceId')
+
+        desired_capacity = asg.describe_auto_scaling_groups(AutoScalingGroupNames=[asg_name])\
+            .get('AutoScalingGroups')[0]\
+            .get('DesiredCapacity')
         config.parameters.append(('InitialQueueSize', desired_capacity))
 
     # Get the MasterSubnetId and use it to determine AvailabilityZone
     try:
         i = [p[0] for p in config.parameters].index('MasterSubnetId')
         master_subnet_id = config.parameters[i][1]
         try:
-            vpcconn = boto.vpc.connect_to_region(config.region,aws_access_key_id=config.aws_access_key_id,
-                                                 aws_secret_access_key=config.aws_secret_access_key)
-            availability_zone = str(vpcconn.get_all_subnets(subnet_ids=master_subnet_id)[0].availability_zone)
-        except boto.exception.BotoServerError as e:
-            logger.critical(e.message)
+            ec2 = boto3.client('ec2', region_name=config.region,
+                               aws_access_key_id=config.aws_access_key_id,
+                               aws_secret_access_key=config.aws_secret_access_key)
+            availability_zone = ec2.describe_subnets(SubnetIds=[master_subnet_id]) \
+                .get('Subnets')[0] \
+                .get('AvailabilityZone')
+        except ClientError as e:
+            logger.critical(e.response.get('Error').get('Message'))
             sys.exit(1)
         config.parameters.append(('AvailabilityZone', availability_zone))
     except ValueError:
         pass
 
     try:
         logger.debug((config.template_url, config.parameters))
-        stack = cfnconn.update_stack(stack_name,template_url=config.template_url,
-                                     parameters=config.parameters, capabilities=capabilities,
-                                     disable_rollback=args.norollback)
-        status = cfnconn.describe_stacks(stack)[0].stack_status
+
+        cfn_params = [{'ParameterKey': param[0], 'ParameterValue': str(param[1])} for param in config.parameters]
+        stack = cfn.update_stack(StackName=stack_name,TemplateURL=config.template_url,
+                                 Parameters=cfn_params, Capabilities=capabilities)
+        status = stack.get('Stacks')[0].get('StackStatus')
```

Comment:
`stack` should be the output of [describe_stacks()](https://boto3.readthedocs.io/en/latest/reference/services/cloudformation.html#CloudFormation.Client.describe_stacks)

Reply:
changed.